packages/flutter/test/material/filter_chip_test.dart

```diff
@@ -938,4 +948,67 @@ void main() {
     // Delete button should be visible.
     expectLater(find.byType(RawChip), matchesGoldenFile('filter_chip.disabled.delete_button.png'));
   });
+
+  testWidgets('Material3 - FilterChip icon colors', (WidgetTester tester) async {
+    const IconData avatarIcon = Icons.house;
+    final ThemeData theme = ThemeData();
+
+    Widget buildFlatChip({
+      required bool selected,
+    }) {
+      return MaterialApp(
+        theme: theme,
+        home: Material(
+          child: Center(
+            child: FilterChip(
+              avatar: const Icon(avatarIcon),
+              onSelected: (bool valueChanged) { },
+              label: const Text('FilterChip'),
+              selected: selected,
+              onDeleted: () { },
+            ),
+          ),
+        ),
+      );
+    }
+
+    Widget buildElevatedChip({
+      required bool selected,
+    }) {
```

Comment:
Why not put both flat and elevated chips in a column?  This will simplify the test

Reply:
Then it would be harder to obtain the correct "clear" icon in the tree.
Currently the following is being used for the expect. Internally it does a `firstWidget` finder, which would be problematic if there are 2 of them.

`getIconDataForIcon(tester, Icons.clear).color`

Reply:
`getIconDataForIcon` is only used by this one widget.  You can move it tin the test and retrieve first and last icon or use different icons